src/encode.rs

```diff
@@ -184,13 +178,13 @@ impl<'a, I: AsRef<[u8]>> EncodeBuilder<'a, I> {
     /// assert_eq!(
     ///     "QuT57JNzzWTu7mW",
     ///     bs58::encode(input)
-    ///         .with_check()
+    ///         .with_check(None)
     ///         .into_string());
     /// ```
     #[cfg(feature = "check")]
     #[cfg_attr(docsrs, doc(cfg(feature = "check")))]
-    pub fn with_check(mut self) -> EncodeBuilder<'a, I> {
-        self.check = Check::Enabled;
+    pub fn with_check(mut self, expected_ver: Option<u8>) -> EncodeBuilder<'a, I> {
```

Comment:
To avoid the breaking change this could be a new function `pub fn with_check_version(self, version: u8)`.